includes/class-give-donor.php

```diff
@@ -278,6 +278,62 @@ public function setup_address() {
 		}
 	}
 
+	/**
+	 * Returns the saved address for a donor
+	 *
+	 * @access public
+	 *
+	 * @since  2.1.3
+	 *
+	 * @param array $args donor address.
+	 *
+	 * @return array The donor's address, if any
+	 */
+	public function get_donor_address( $args = array() ) {
+		$args = wp_parse_args(
+			$args,
+			array(
+				'address_type' => 'billing',
+			)
+		);
+
+		$default_address = array(
+			'line1'   => '',
+			'line2'   => '',
+			'city'    => '',
+			'state'   => '',
+			'country' => '',
+			'zip'     => '',
+		);
+
+		// Backward compatibility.
+		if ( ! give_has_upgrade_completed( 'v20_upgrades_user_address' ) ) {
+
+			// Backward compatibility for user id param.
+			$by_user_id = get_user_by( 'id', $this->id ) ? true : false;
+
+			if ( $by_user_id ) {
+				return wp_parse_args( (array) get_user_meta( $this->id, '_give_user_address', true ), $default_address );
```

Comment:
@raftaar1191 it will be user id instead of donor id

Reply:
PR update @ravinderk as we have discuss over the call